src/annotator/components/ProfileModal.tsx

```diff
@@ -3,35 +3,51 @@ import classnames from 'classnames';
 import { CancelIcon, IconButton } from '@hypothesis/frontend-shared/lib/next';
 
 import type { Emitter, EventBus } from '../util/emitter';
+import type { FeatureFlags } from '../features';
 
 export type ProfileConfig = { profileAppUrl: string } & Record<string, unknown>;
 
 type ProfileModalProps = {
   eventBus: EventBus;
   config: ProfileConfig;
+  features: FeatureFlags;
 };
 
-export function ProfileModal({ eventBus, config }: ProfileModalProps) {
+export function ProfileModal({
+  eventBus,
+  config,
+  features,
+}: ProfileModalProps) {
   const [isHidden, setIsHidden] = useState(true);
   const emitterRef = useRef<Emitter | null>(null);
+  // Used only to track when was this modal first open, delaying the iframe to
+  // be loaded until strictly necessary.
+  const [hasOpened, setHasOpened] = useState(false);
 
   useEffect(() => {
     const emitter = eventBus.createEmitter();
     emitter.subscribe('openProfile', () => {
```

Comment:
I would assume the menu item that triggers this event is not visible if the feature flag is not enabled? If so then it seems to me we don't actually need to consult the feature flag here, we can just assume that if we receive this event, then the flag is enabled. You can still use the `hasOpened`/`setHasOpened` logic to defer loading the iframe until the first time `openProfile` is received.

Reply:
Yeah, you are right there, and one of the options I considered was that (see the third scratched-out point, in the `Possible next steps:` section of this PR's description).

However, it feels like we are building the component in the assumption that this is taken care of by something else, instead of letting it take care of everything on itself. So I don't know, I have mixed feelings 😅 

If you think the right thing to do is just simplify everything, I'll do the changes.

Reply:
Fair enough, you are probably right. I'll change this to siply handle the state internally and not rely on the feature flag.